core/src/main/java/org/jruby/javasupport/ext/JavaExtensions.java

```diff
@@ -0,0 +1,37 @@
+package org.jruby.javasupport.ext;
+
+import org.jruby.Ruby;
+import org.jruby.RubyModule;
+import org.jruby.util.cli.Options;
+
+import java.util.function.Consumer;
+
+/**
+ * Lazy Java class extensions initialization.
+ *
+ * @note Internal API
+ * @author kares
+ */
+public class JavaExtensions {
+
+    private static final boolean LAZY = Options.JI_LOAD_LAZY.load();;
+
+    private JavaExtensions() { /* hidden */ }
+
+    static void put(final Ruby runtime, Class javaClass, Consumer<RubyModule> proxyClass) {
+        if (!LAZY) {
+            proxyClass.accept( org.jruby.javasupport.Java.getProxyClass(runtime, javaClass) );
+            return;
+        }
+        Object previous = runtime.getJavaExtensionDefinitions().put(javaClass, proxyClass);
+        assert previous == null;
+    }
+
+    public static void define(final Class javaClass, final RubyModule proxyClass) {
```

Comment:
Seems like runtime could be a parameter here to match put().

Reply:
It also seems to line up more with Iterable stuff below which also have runtime.